src/System.Net.Security/src/System/Net/SecureProtocols/NegoState.Windows.cs

```diff
@@ -13,869 +13,614 @@
 namespace System.Net.Security
 {
     //
-    // The class maintains the state of the authentication process and the security context.
-    // It encapsulates security context and does the real work in authentication and
+    // The class does the real work in authentication and
     // user data encryption with NEGO SSPI package.
     //
     // This is part of the NegotiateStream PAL.
     //
-    internal class NegoState
+    internal partial class NegoState
     {
-        private const int ERROR_TRUST_FAILURE = 1790;   // Used to serialize protectionLevel or impersonationLevel mismatch error to the remote side.
-
-        private static readonly byte[] s_emptyMessage = new byte[0];
-        private static readonly AsyncCallback s_readCallback = new AsyncCallback(ReadCallback);
-        private static readonly AsyncCallback s_writeCallback = new AsyncCallback(WriteCallback);
-
-        private Stream _innerStream;
-        private bool _leaveStreamOpen;
+        internal IIdentity GetIdentity()
+        {
+            CheckThrow(true);
 
-        private Exception _exception;
+            IIdentity result = null;
+            string name = _context.IsServer ? _context.AssociatedName : _context.Spn;
+            string protocol = "NTLM";
 
-        private StreamFramer _framer;
-        private NTAuthentication _context;
+            protocol = _context.ProtocolName;
 
-        private int _nestedAuth;
+            if (_context.IsServer)
+            {
+                SecurityContextTokenHandle token = null;
+                try
+                {
+                    //token = _context.GetContextToken();
```

Comment:
Yes.. the comment needs to be removed. Since token can be fetched only on Windows, the code had to be refactored. On Unix, this will throw PlatformNotSupported in v1 since server support won't be there